.github/actions/upload/action.yml

```diff
@@ -0,0 +1,43 @@
+name: upload
+description: "Upload test results, reports, and heap dump"
+inputs:
+  name-suffix:
+    default: ${{ github.job }}
+  test-report-retention-days:
+    default: 5
+  heap-dump-retention-days:
+    default: 2
+  upload-heap-dump:
+    default: "false"
```

Comment:
stringly typed booleans create confusion specially when the value being used can be easily confused for a boolean